### PR TITLE
100% code coverage

### PR DESCRIPTION
**Pull request by Tpt**

### CODE REVIEW

**Comment by Tpt** [comment]
@rubensworks Could you add this repository to Coveralls?

**Comment by rubensworks** [comment]
AFAIK, this isn't needed (anymore).

From the action's readme:

> You don't need to add the repo to Coveralls first, it will be created when receiving the post.

I think [this part](https://github.com/comunica/comunica/blob/master/.github/workflows/ci.yml#L39-L44) of the CI file is missing.

**Comment by Tpt** [comment]
@rubensworks Thank you. It's now fixed.

File: lib/JsonEventParser.ts [hunk 1]
```diff
@@ -98,7 +98,8 @@ export class JsonEventParser extends Transform {
         return key;
       }
     }
-    return `0x${code.toString(16)}`;
+    /* istanbul ignore next */
+    throw new Error(`Not supported token code ${code}`);
```
Comment (rubensworks) [inline lib/JsonEventParser.ts:102]: Not a big fan of skipping coverage lines 😅 
Could we add a small test for this case?

File: lib/JsonEventParser.ts [hunk 2]
```diff
@@ -460,6 +445,7 @@ export class JsonEventParser extends Transform {
 
   private popFromStack(token: number): void {
     const parent = this.stack.pop();
+    /* istanbul ignore next */
```
Comment (rubensworks) [inline lib/JsonEventParser.ts:448]: Also here?

**Review by rubensworks — CHANGES_REQUESTED** [review]

File: lib/JsonEventParser.ts (hunk 1 above)
Comment (Tpt) [reply lib/JsonEventParser.ts:102]: This line is here to make the TS compiler happy by returning a string always but can't be evaluated anywhere (the method is private and called only with valid error codes). To cover it I would have to make it public or hack around the TypeScript compiler. I have not found a proper way to state that a piece of code is unreachable in TS or state in the tpye system that only keys of a given object are allowed.

Comment (rubensworks) [reply lib/JsonEventParser.ts:102]: > To cover it I would have to make it public or hack around the TypeScript compiler. I have not found a proper way to state that a piece of code is unreachable in TS or state in the tpye system that only keys of a given object are allowed.

I guess casting the parser to `any` and invoking the private method could work. (or might have to become `protected`).

File: lib/JsonEventParser.ts (hunk 2 above)
Comment (rubensworks) [reply lib/JsonEventParser.ts:448]: I would definitely keep it indeed. Should be possible to unit test it by casting the parser to `any` like above.

Comment (Tpt) [reply lib/JsonEventParser.ts]: Done. Thanks!

**Comment by rubensworks** [comment]
Super! Released as `1.0.0-beta.3`.

**Pull request merged**